### PR TITLE
add support for signing

### PR DESCRIPTION
**Pull request by amandeepgautam**
Add support for signing. Related discussion: https://github.com/simo5/gss-ntlmssp/issues/2

### CODE REVIEW

File: src/gss_sec_ctx.c [hunk 1]
```diff
@@ -1263,6 +1263,34 @@ static uint32_t gssntlm_sasl_ssf(uint32_t *minor_status,
     return GSSERRS(retmin, retmaj);
 }
 
+/* GSS_C_INQ_SSPI_SESSION_KEY 1.2.840.113554.1.2.2.5.5 */
+static const gss_OID_desc sspi_session_key_oid = {
+  11, discard_const("\x2a\x86\x48\x86\xf7\x12\x01\x02\x02\x05\x05")
+};
```
Comment (simo5) [inline src/gss_sec_ctx.c]: This OID is already available in gssapi_ext.h as GSS_C_INQ_SSPI_SESSION_KEY so we do not need to redeclare it here.

File: src/gss_sec_ctx.c [hunk 2]
```diff
@@ -1263,6 +1263,34 @@ static uint32_t gssntlm_sasl_ssf(uint32_t *minor_status,
...
+                                         struct gssntlm_ctx *ctx,
+                                         gss_buffer_set_t *data_set)
+{
+    uint32_t retmin;
+    uint32_t retmaj;
+    uint32_t tmpmin;
+    gss_buffer_desc session_key_buf;
+
+    if (ctx->exported_session_key.length == 0) {
+      return GSSERRS(ERR_NOTSUPPORTED, GSS_S_UNAVAILABLE);
```
Comment (simo5) [inline src/gss_sec_ctx.c]: This should use ERR_NOTAVAIL

**Comment by simo5** [comment]
Ah please provide a better name for the commit, and a description of what it is used for.
(Linux kernel formatting stile for git comment subject and comments)

**Comment by amandeepgautam** [comment]
@simo5 addressed the comments.

File: src/gss_sec_ctx.c [hunk 3]
```diff
@@ -1289,6 +1312,8 @@ uint32_t gssntlm_inquire_sec_context_by_oid(uint32_t *minor_status,
         return gssntlm_spnego_req_mic(minor_status, ctx, data_set);
     } else if (gss_oid_equal(desired_object, &sasl_ssf_oid)){
         return gssntlm_sasl_ssf(minor_status, ctx, data_set);
+    } else if (gss_oid_equal(desired_object, &sspi_session_key_oid)) {
```
Comment (simo5) [inline src/gss_sec_ctx.c]: Shouldn't this be changed to use GSS_C_INQ_SSPI_SESSION_KEY?

Comment (amandeepgautam) [reply src/gss_sec_ctx.c]: done.

**Comment by simo5** [comment]
Thank you Aman

**Comment by amandeepgautam** [comment]
@simo5  Thanks for your help in resolving this.

**Pull request merged**